src/Features/CSharp/Portable/Completion/CompletionProviders/DeclarationNameCompletionProvider.cs

```diff
@@ -177,7 +177,7 @@ private ITypeSymbol UnwrapType(ITypeSymbol type, Compilation compilation)
                     {
                         foreach (var baseName in baseNames)
                         {
-                            var name = rule.NamingStyle.CreateName(baseName);
+                            var name = EscapeIdentifier(rule.NamingStyle.CreateName(baseName));
```

Comment:
We have a helper for this: (http://source.roslyn.io/#Microsoft.CodeAnalysis.CSharp.Workspaces/Extensions/StringExtensions.cs,14). You should be able to get the "isQueryContext" parameter from the CSharpSyntaxContext that's passed in.